### PR TITLE
NO-ISSUE: Migrate build image to use registry.access.redhat.com/ubi9/…

### PR DESCRIPTION
**Pull request by pastequo**
…go-toolset:1.20

Another try of https://github.com/openshift/assisted-installer-agent/pull/734

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -76,7 +76,7 @@ endif
 unit-test:
 	$(MAKE) _test TEST_SCENARIO=unit TIMEOUT=30m TEST="$(or $(TEST),$(shell go list ./... | grep -v subsystem))" || (docker kill postgres && /bin/false)
 
-subsystem: build-image
```
Comment (danmanor) [inline Makefile:79]: why ?

**Comment by danmanor** [comment]
overriding these jobs as they are not affected

**Comment by danmanor** [comment]
/override ci/prow/e2e-agent-compact-ipv4 ci/prow/edge-e2e-ai-operator-ztp ci/prow/edge-e2e-metal-assisted

File: Makefile (hunk 1 above)
Comment (pastequo) [reply Makefile:79]: good catch this was to speed up my local test

File: subsystem/ntp_test.go [hunk 2]
```diff
@@ -180,5 +180,5 @@ func getNTPResponseFromStepReply(actualReply *models.StepReply) *models.NtpSynch
 func addChronyDaemon(ctx context.Context, hostID string) {
 	s, e, errorCode := agentUtils.Execute("docker", []string{"exec", "agent", "chronyd"}...)
 	fmt.Println(s, e, errorCode)
-	Expect(errorCode).To(Equal(0))
```
[the remark this answers is not in the document]
Comment (pastequo) [reply subsystem/ntp_test.go:183]: I can remove it, but our log are terrible when a test fail (because of our usage of ginkgo). Like

```
• Failure [5.287 seconds]
NTP tests
/home/assisted/subsystem/ntp_test.go:23
  add_new_server
  /home/assisted/subsystem/ntp_test.go:39
    IP [It]
    /home/assisted/subsystem/ntp_test.go:40

    Expected

              
        <int>: 125
    to equal
        <int>: 0
```

Hard to see what exactly failed or was checked

[the remark this answers is not in the document]
Comment (danmanor) [reply subsystem/ntp_test.go:183]: Oh this is an explanation on the failure, nice

[the remark this answers is not in the document]
Comment (danmanor) [reply subsystem/ntp_test.go:183]: btw - what caused this test to fail ?

[the remark this answers is not in the document]
Comment (pastequo) [reply subsystem/ntp_test.go:183]: 1. `dnf install docker` installed podman
2. We start the "test infra" with `docker-compose` using docker below
3. We run `docker exec xxx` using podman below
4. Podman doesn't see container started by docker-compose
5. Exec failed

**Comment by danmanor** [comment]
/override ci/prow/e2e-agent-compact-ipv4 ci/prow/edge-e2e-ai-operator-ztp ci/prow/edge-e2e-metal-assisted

File: Dockerfile.assisted_installer_agent-build [hunk 3]
```diff
@@ -1,9 +1,20 @@
-FROM registry.ci.openshift.org/openshift/release:golang-1.20
-ENV GO111MODULE=on
```
Comment (danmanor) [inline Dockerfile.assisted_installer_agent-build:2]: BTW - @pastequo why did you remove this ?

Comment (pastequo) [reply Dockerfile.assisted_installer_agent-build:2]: because it's default value in 1.20
